code/modules/reagents/Chemistry-Goon-Pyrotechnics.dm

```diff
@@ -160,9 +179,279 @@
 /datum/chemical_reaction/blackpowder_explosion/on_reaction(var/datum/reagents/holder, var/created_volume)
 	sleep(rand(50,100))
 	var/turf/simulated/T = get_turf(holder.my_atom)
-	var/ex_severe = round(created_volume / 10)
-	var/ex_heavy = round(created_volume / 8)
-	var/ex_light = round(created_volume / 6)
-	var/ex_flash = round(created_volume / 4)
+	var/ex_severe = round(created_volume / 100)
+	var/ex_heavy = round(created_volume / 42)
+	var/ex_light = round(created_volume / 21)
+	var/ex_flash = round(created_volume / 8)
+	explosion(T,ex_severe,ex_heavy,ex_light,ex_flash, 1)
+	return
+
+/datum/reagent/flash_powder
+	name = "Flash Powder"
+	id = "flash_powder"
+	description = "Makes a very bright flash."
+	reagent_state = LIQUID
+	color = "#000000"  //rgb: 96, 165, 132
+
+/datum/chemical_reaction/flash_powder
+	name = "Flash powder"
+	id = "flash_powder"
+	result = "flash_powder"
+	required_reagents = list("aluminium" = 1, "potassium" = 1, "sulfur" = 1 )
+	result_amount = 3
+
+/datum/chemical_reaction/flash_powder_flash
+	name = "Flash powder activation"
+	id = "flash_powder_flash"
+	result = null
+	required_reagents = list("flash_powder" = 1)
+	required_temp = 374
+
+/datum/chemical_reaction/flash_powder_flash/on_reaction(var/datum/reagents/holder, var/created_volume)
+	var/location = get_turf(holder.my_atom)
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+	s.set_up(2, 1, location)
+	s.start()
+	for(var/mob/living/carbon/C in get_hearers_in_view(5, location))
+		if(C.eyecheck())
+			continue
+		flick("e_flash", C.flash)
+		if(get_dist(C, location) < 4)
+			C.Weaken(5)
+			continue
+		C.Stun(5)
+
+/datum/chemical_reaction/flash_powder/on_reaction(var/datum/reagents/holder, var/created_volume)
+	if(holder.has_reagent("stabilizing_agent"))
+		return
+	var/location = get_turf(holder.my_atom)
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+	s.set_up(2, 1, location)
+	s.start()
+	for(var/mob/living/carbon/C in get_hearers_in_view(5, location))
+		if(C.eyecheck())
+			continue
+		flick("e_flash", C.flash)
+		if(get_dist(C, location) < 4)
+			C.Weaken(5)
+			continue
+		C.Stun(5)
+	holder.remove_reagent("flash_powder", created_volume)
+
+/datum/reagent/smoke_powder
+	name = "Smoke Powder"
+	id = "smoke_powder"
+	description = "Makes a large cloud of smoke that can carry reagents."
```

Comment:
This is the style of description you want to be going for - it describes the events in in-universe terms while being absolutely clear about what it does.

And example of this done badly would be "Creates 6 smokecloud objects that apply INGEST reactions with all contained reagents